File: observer/Student.java
package observer;

public class Student implements Observer {
    private String _name;
    private String _tPhone;

    public String get_name() {
        return _name;
    }

    public void set_name(String _name) {
        this._name = _name;
    }

    public String get_tPhone() {
        return _tPhone;
    }

    public void set_tPhone(String _tPhone) {
        this._tPhone = _tPhone;
    }

    public Student(String name) {
        set_name(name);
    }

    @Override
    public void update(Object o) {
        set_tPhone((String)o);
    }

    public void show() {
        System.out.println("Name: " +  get_name() + "\nTeacher's Phone: " + get_tPhone());
    }
}

File: Js/observer/Student.js
var Student = /** @class */ (function () {
    function Student(name) {
        this.set_name(name);
    }
    Student.prototype.set_name = function (name) {
        this._name = name;
    };
    Student.prototype.get_name = function () {
        return this._name;
    };
    Student.prototype.set_tPhone = function (tPhone) {
        this._tPhone = tPhone;
    };
    Student.prototype.get_tPhone = function () {
        return this._tPhone;
    };
    Student.prototype.update = function (o) {
        this.set_tPhone(o);
    };
    Student.prototype.show = function () {
        console.log("Name: " + this.get_name() + "\nTeacher's Phone: " + this.get_tPhone());
    };
    return Student;
}());
